src/renderer/modals/loadingScreen/index.tsx

```diff
@@ -1,65 +1,60 @@
 // Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React from 'react';
+import React, {useState, useEffect} from 'react';
 import ReactDOM from 'react-dom';
 
+import {useConfig} from 'renderer/hooks/useConfig';
+
 import LoadingScreen from '../../components/LoadingScreen';
 
 import 'renderer/css/components/LoadingAnimation.css';
 import 'renderer/css/components/LoadingScreen.css';
 
-type Props = Record<string, never>;
-
-type State = {
-    showLoadingScreen: boolean;
-    darkMode: boolean;
-}
+const LoadingScreenRoot: React.FC = () => {
+    const [showLoadingScreen, setShowLoadingScreen] = useState(true);
+    const {config} = useConfig();
 
-class LoadingScreenRoot extends React.PureComponent<Props, State> {
-    constructor(props: Props) {
-        super(props);
-        this.state = {
-            showLoadingScreen: true,
-            darkMode: false,
+    useEffect(() => {
+        const handleToggleLoadingScreenVisibility = (showLoadingScreen: boolean) => {
+            setShowLoadingScreen(showLoadingScreen);
         };
-    }
-
-    async componentDidMount() {
-        window.desktop.onDarkModeChange(this.setDarkMode);
-        const darkMode = await window.desktop.getDarkMode();
-        this.setDarkMode(darkMode);
 
-        window.desktop.loadingScreen.onToggleLoadingScreenVisibility(this.onToggleLoadingScreenVisibility);
+        const initializeApp = async () => {
+            window.desktop.loadingScreen.onToggleLoadingScreenVisibility(handleToggleLoadingScreenVisibility);
 
-        window.addEventListener('click', () => {
-            window.desktop.closeServersDropdown();
-            window.desktop.closeDownloadsDropdown();
-        });
-    }
+            window.addEventListener('click', () => {
+                window.desktop.closeServersDropdown();
+                window.desktop.closeDownloadsDropdown();
+            });
+        };
```

Comment:
This lack consistency. You are putting one of the functions (the one for click) inline, while the other, (handle toggle visibility) you put it in a separate function.

I have no strong preference for any of them, but we should be consistent.

Reply:
Sure.